service/notesservice.php

```diff
@@ -138,15 +138,31 @@ public function update ($id, $content, $userId, $mtime=0) {
         // using a maximum of 100 chars should be enough
         $title = mb_substr($title, 0, 100, "UTF-8");
 
-        // generate filename if there were collisions
-        $currentFilePath = $file->getPath();
-        $basePath = pathinfo($file->getPath(), PATHINFO_DIRNAME);
-        $fileExtension = pathinfo($file->getName(), PATHINFO_EXTENSION);
-        $newFilePath = $basePath . '/' . $this->generateFileName($folder, $title, $fileExtension, $id);
 
-        // if the current path is not the new path, the file has to be renamed
-        if($currentFilePath !== $newFilePath) {
-            $file->move($newFilePath);
+        // rename/move file with respect to title/category
+        // this can fail if access rights are not sufficient or category name is illegal
+        try {
+            $currentFilePath = $file->getPath();
+            $fileExtension = pathinfo($file->getName(), PATHINFO_EXTENSION);
+
+            // detect (new) folder path based on category name
+            if($category===null) {
+                $basePath = pathinfo($file->getPath(), PATHINFO_DIRNAME);
+            } else {
+                $basePath = $notesFolder->getPath();
+                if(!empty($category))
+                    $basePath .= '/'.$category;
+                $this->getOrCreateFolder($basePath);
+            }
+
+            // assemble new file path
+            $newFilePath = $basePath . '/' . $this->generateFileName($folder, $title, $fileExtension, $id);
+
+            // if the current path is not the new path, the file has to be renamed
+            if($currentFilePath !== $newFilePath) {
+                $file->move($newFilePath);
+            }
+        } catch(\Exception $e) {
         }
```

Comment:
Can we specify a stricter Exception than the generic `\Exception`?
Also it would be great to at least give some kind of feedback in form of a server log entry.